src/playground-project.ts

```diff
@@ -270,12 +295,18 @@ export class PlaygroundProject extends LitElement {
         };
       })
     );
-    this._compileProject();
+    if (!this._filesSetExternally) {
```

Comment:
should this check come earlier? It looks like if `this._filesSetExternally` is true you fetch the files then throw them away

Reply:
Just being defensive in case files was explicitly set after the project fetch started, but before it completed, to make sure the "explicitly set always wins" rule always holds. Added a comment.

Reply:
Oh and, it is also checked before the fetch starts.

Reply:
ah, ok